src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EditorConfig/LanguageExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.EditorConfig;
+using static Microsoft.CodeAnalysis.EditorConfig.LanguageConstants;
+
+internal static class LanguageExtensions
+{
+    public static Language GetLanguageFromFilePath(this string filePath)
+    {
+        var language = default(Language);
+        var fileExtension = Path.GetExtension(filePath);
+        if (fileExtension == DefaultCSharpExtension)
+        {
+            language = Language.CSharp;
+        }
+
+        if (fileExtension == DefaultVisualBasicExtension)
+        {
+            language = Language.VisualBasic;
+        }
+
+        return language;
```

Comment:
For `Language` the values are `CSharp = 1` and `VisualBasic = 2`. The implication here of returning `default(Language)` is that .HasFlags always returns false since its value is `0`. My indention was that we wouldn't throw if someone asked us about a language we didn't understand but that may be too subtle. If we want it to fail than I would change it to `TryGetLanguageFromFilePath` to explain that this operation does not always succeed.